overlord/devicestate/devicemgr.go

```diff
@@ -1041,6 +1041,57 @@ func (m *DeviceManager) ensureAutoImportAssertions() error {
 	return nil
 }
 
+func (m *DeviceManager) ensureSerialBoundAssertionsProcessed() error {
+	if release.OnClassic {
+		return nil
+	}
+
+	if m.earlyDeviceSeed != nil {
+		return nil
+	}
+
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	var seeded bool
+	if err := m.state.Get("seeded", &seeded); err != nil && !errors.Is(err, state.ErrNoState) {
+		return err
+	}
+	if !seeded {
+		return nil
+	}
+
+	var waitingOnSerial bool
+	err := m.state.Get("assertion-waiting-on-serial", &waitingOnSerial)
+	if err != nil && !errors.Is(err, state.ErrNoState) {
+		logger.Noticef("failed to get serial-bound assert state")
+	}
+	if !waitingOnSerial {
+		return nil
+	}
+
+	model, err := m.Model()
+	if err != nil {
+		return nil
+	}
+
+	serial, err := m.Serial()
```

Comment:
I think this should probably propogate an error returned here *some of the time*. The error handling on line 604 should apply here too

Reply:
hmm.. this should probably just return the error all the time, actually. There's a prior segment of this function which checks if the device is seeded. If we're seeded and don't have a serial for any reason, that seems worthy of throwing an error.